packages/node/src/indexer/indexer.manager.ts

```diff
@@ -290,7 +290,7 @@ export class IndexerManager {
     }
   }
 
-  private async indexData<K extends SubstrateHandlerKind>(
+  async indexData<K extends SubstrateHandlerKind>(
```

Comment:
We should leave this private and use indexBlock

Reply:
how shall we get the datasource for the test? maybe we can pass a ds filter function as an optional parameter to indexBlock that should do additional filtering on top of datasources obtained from projectService.

```
 async indexBlock(
    blockContent: BlockContent,
    runtimeVersion: RuntimeVersion,
    filterDs?: (ds: SubqlProjectDs[]) => SubqlProjectDs[]
  ) {
       
       
       if (filterDs) {
        datasources = filterDs(datasources);
      }
  }
```